Sources/DDGSync/DDGSync.swift

```diff
@@ -241,29 +242,30 @@ public class DDGSync: DDGSyncing {
 
         authState = account?.state ?? .inactive
 
+        guard let account else {
+            do {
+                try removeAccount(reason: .notFoundInSecureStorage)
+            } catch {
+                dependencies.errorEvents.fire(.failedToRemoveAccount, error: error)
+            }
+            return
+        }
+
         do {
             try updateAccount(account)
         } catch {
             dependencies.errorEvents.fire(.failedToSetupEngine, error: error)
         }
     }
 
-    private func updateAccount(_ account: SyncAccount? = nil) throws {
-        guard account?.state != .initializing else {
+    private func updateAccount(_ account: SyncAccount) throws {
```

Comment:
Differentiating this from account removal to:
- Enforce giving a reason for reporting
- Prevent propagating nil values resulting in account removal